### ISSUE TITLE
Embedded links doesn't execute

### ISSUE THREAD

**Issue by phamdacloc** [WP-API/WP-API#1645]
I'm reading the WP-REST documentation on [Embedding link](http://v2.wp-api.org/extending/linking/) but having a hard time getting it to work; either my understanding is incorrect or the 'embeddable' attribute is broken.

I'm creating a new REST endpoint that has this route: 'http://localhost:8888/wp-json/bbpress_ext/v1/author/1`

The endpoint is coded like this:

```php
function my_awesome_func( $data ) {
	
	$tmp = array( 'aaa' => "123",
				  'bbb' => '456',
				  'ccc' => '789');
	
	$response = new WP_REST_Response($tmp);
	$response->add_link( 'author', rest_url( '/wp/v2/users/1' ), array( 'embeddable' => true ) );

	return $response;
}

add_action( 'rest_api_init', function () {
    register_rest_route( 'bbpress_ext/v1', '/author/(?P<id>\d+)', array(
        'methods' => 'GET',
        'callback' => 'my_awesome_func',
    ) );
} );
```

I'm getting the following RESPONSE when I execute the new route: `http://localhost:8888/wp-json/bbpress_ext/v1/author/1`
```php
{
  "aaa": "123",
  "bbb": "456",
  "ccc": "789",
  "_links": {
    "author": [
      {
        "embeddable": true,
        "href": "http://localhost:8888/wp-json/wp/v2/users/1"
      }
    ]
  }
}
```

I'm EXPECTING the following from the response:
```php
{
  "aaa": "123",
  "bbb": "456",
  "ccc": "789",
  "_links": {
    "author": [
      {
        "embeddable": true,
        
        {
             "avatar_urls": {
             "24": "http://2.gravatar.com/avatar/29b3ef85f13fedb43f84e6cb4a634e73?s=24&d=mm&r=g",
             "48": "http://2.gravatar.com/avatar/29b3ef85f13fedb43f84e6cb4a634e73?s=48&d=mm&r=g",
             "96": "http://2.gravatar.com/avatar/29b3ef85f13fedb43f84e6cb4a634e73?s=96&d=mm&r=g"
        },
        "description": "Just another developer.",
        "id": 1,
        "link": "http://localhost:8888/author/ninsutuadmin/",
        "name": "ninsutuAdmin",
        "url": "http://mysite.com",
        "userMeta": null,
        "_links": {
        "self": [
                  {
                      "href": "http://localhost:8888/wp-json/wp/v2/users/1"
                  }
                  ],
         "collection": [
                   {
                         "href": "http://localhost:8888/wp-json/wp/v2/users"
                   }
              ]
         }
      }

      }
    ]
  }
}
```

Basically, I'm expecting `wp/v2/users/1` to have executed and responded with its data. Is my understand way off?

**Comment by rmccue** [comment]
You need to add `?_embed` to the request to get the embedded data (i.e. clients need to opt-in to getting embeds).

**Comment by phamdacloc** [comment]
Thanks, I read that in the documentation but couldn't grasp it piece of info.
